src/alias/azext_alias/alias.py

```diff
@@ -131,25 +131,29 @@ def transform(self, args):
         """
         if self.parse_error():
             # Write an empty hash so next run will check the config file against the entire command table again
-            self.write_alias_config_hash(True)
+            AliasManager.write_alias_config_hash(empty_hash=True)
             return args
 
         # Only load the entire command table if it detects changes in the alias config
         if self.detect_alias_config_change():
             self.load_full_command_table()
-            self.build_collision_table()
+            self.collided_alias = AliasManager.build_collision_table(self.alias_table.sections(),
+                                                                     self.reserved_commands)
         else:
             self.load_collided_alias()
 
         transformed_commands = []
         alias_iter = enumerate(args, 1)
-
+        ignore_next_iter = False
         for alias_index, alias in alias_iter:
             # Directly append invalid alias or collided alias
-            if not alias or alias[0] == '-' or (alias in self.collided_alias and
-                                                alias_index in self.collided_alias[alias]):
+            if not alias or alias[0] == '-' or ignore_next_iter or (alias in self.collided_alias and
+                                                                    alias_index in self.collided_alias[alias]):
                 transformed_commands.append(alias)
+                ignore_next_iter = alias and alias[0] == '-'
                 continue
+            else:
+                ignore_next_iter = False
```

Comment:
Could you elaborate the logic here in the comments?

Reply:
The logic here is to not transform named arguments. For example, if you have an alias `ls` pointing to `list`, running `az vm create -g xxx -n ls` should not yield `az vm create -g xxx -n list`,
 I think I can make it less ugly and more readable.